.github/workflows/documentation.yml

```diff
@@ -0,0 +1,47 @@
+name: Documentation
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  api-documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Set up Doxygen
+        run: sudo apt-get install -y doxygen
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Set up dependency cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('doc/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        working-directory: ./doc
+        run: pip install -r requirements.txt
+
+      - name: Build documentation
+        working-directory: ./doc
+        run: make html
+
+      - name: Upload documentation
+        uses: actions/upload-artifact@v2
+        with:
+          name: api-docs
+          path: ./doc/build/html
```

Comment:
I think this produces an archive without the parent directory, correct? If so, could you add a comment that `api-docs.zip` will only include the content of `./doc/build/html` without the parent directory?

Reply:
Correct - it's the HTML output of the sphinx only.

A common the the YAML would suffice?

Reply:
Yes, please add a comment to the .yml file, for example above the step name. Thanks!